cookbooks/stac-zarr-report/data-producers/index.qmd

```diff
@@ -0,0 +1,184 @@
+
+This is similar to **One big Zarr store - minimal STAC metadata** with the exception that data does not need to be stored in a Zarr store. It can be stored in any number of COG or NetCDF or HDF5 files. Anything that has consistent chunking, encoding, and aligned coordinates on-disc.
+
+1. Create a virtual reference file (kerchunk) or store (icechunk) pointing to chunks of data on-disc.
+2. Define a STAC collection for each virtual dataset.
+3. In each collection include a collection-level asset containing the href link to the virtual reference file.
+
+**Pros**
+- There is no metadata duplication in STAC, so the STAC side is easy to maintain.
+- Simple access interface for Python users - no client-side concatenation.
+- Entire data cube can be lazily constructed with one GET to the reference file or store.
+
+**Cons**
+- Data variables are not exposed at the STAC level, so users cannot discover relevant datasets by searching for variables.
+- Virtual reference file needs to be kept in sync with updates in the underlying data which can be expensive to maintain.
+- Non-Python access is less well developed.
+
+### Virtual dataset in an external file - more STAC metadata
+
+Same as above but with the additional step:
+
+4. Write STAC item metadata using the data cube extension.
+
+**Pros**
+- Data variables are exposed at the STAC level, so users can discover relevant datasets by searching for variables.
+- Simple access interface for Python users - no client-side concatenation.
+- Entire data cube can be lazily constructed with one GET to the reference file.
+
+**Cons**
+- Metadata is duplicated, but it is variable-level so unlikely to change often.
+- Virtual reference file needs to be kept in-sync with updates in the underlying data which can be expensive to maintain.
+- Non-Python access is less well developed.
+
+### Virtual dataset in STAC - most STAC metadata
+
+This is the most experimental approach. It is similar to the one above but instead of a separate reference file the whole chunk manifest is contained within STAC.
+
+1. Create virtual references (kerchunk) pointing to chunks of data on-disc.
+2. Define a STAC collection for each virtual dataset.
+3. Define a STAC item for each spatial temporal chunk within that dataset.
+4. In each STAC item include one asset for each variable with an href link to a chunk of data on-disc and a property containing the kerchunk reference.
+
+**Pros**
+- Simple access interface for Python users - no client-side concatenation.
+- Data variables are exposed at the STAC level, so users can discover relevant datasets by searching for variables.
+- Entire data cube can be lazily constructed directly from STAC response.
+
+**Cons**
+- This is the most verbose option, so it likely to only work when using a STAC API or stac-geoparquet.
+- Virtual references needs to be kept in-sync with updates in the underlying data which can be expensive to maintain.
```

Comment:
This seems hard, tbh. You're saying we'd be doing a full-rewrite of the STAC catalog on every modification to the Zarr store, right?

Reply:
I don't think it would have to be a full rewrite, just the items that changed? In this scenario the actual data can be stored in any chunked format (zarr, netcdf, hdf).